### PR TITLE
`azurerm_monitor_metric_alert`: Removed maxItems for the `scope` property

### LINKED ISSUE #3719: Allow multiple scopes to be passed to Monitor MetricAlert resource
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Affected Resource(s)

* `azurerm_monitor_metric_alert`

### Terraform Configuration Files
N/A

### Debug Output
N/A

### Panic Output
N/A

### Expected Behavior

Multiple `scopes` can be specified for a monitor metric alert resource.

### Actual Behavior

Only one scope is allowed due to an issue in the MS Go SDK, which has since been resolved (see References section below).

### Steps to Reproduce

```hcl
...
...
resource "azurerm_monitor_metric_alert" "test" {
  name                = "example-metricalert"
  resource_group_name = "${azurerm_resource_group.main.name}"
  // Cannot set more than one value here because `MaxItems` is set to 1.
  scopes              = ["${azurerm_storage_account.to_monitor.id}"]
  description         = "Action will be triggered when Transactions count is greater than 50."

  criteria {
    metric_namespace = "Microsoft.Storage/storageAccounts"
    metric_name      = "Transactions"
    aggregation      = "Total"
    operator         = "GreaterThan"
    threshold        = 50

    dimension {
      name     = "ApiName"
      operator = "Include"
      values   = ["*"]
    }
  }

  action {
    action_group_id = "${azurerm_monitor_action_group.main.id}"
  }
}
...
...
```

### Important Factoids
N/A

### References

The relevant line in the resource is: https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/azurerm/resource_arm_monitor_metric_alert.go#L47.

The GitHub [issue](https://github.com/Azure/azure-sdk-for-go/issues/2920) linked in the comment for `scopes` has since been closed as resolved.

### PR DESCRIPTION
**Pull request by e96wic**
Fixes #3719

### CODE REVIEW

File: azurerm/resource_arm_monitor_metric_alert.go [hunk 1]
```diff
@@ -37,16 +37,25 @@ func resourceArmMonitorMetricAlert() *schema.Resource {
...
 
-			// TODO: Multiple resource IDs (Remove MaxItems) support is missing in SDK
-			// Issue to track: https://github.com/Azure/azure-sdk-for-go/issues/2920
-			// But to prevent potential state migration in the future, let's stick to use Set now
 			"scopes": {
 				Type:     schema.TypeSet,
 				Required: true,
 				MinItems: 1,
-				MaxItems: 1,
+				ForceNew: true,
```
Comment (e96wic) [inline azurerm/resource_arm_monitor_metric_alert.go]: we only need to recreate the resource in some occasions, is it possible to make this conditional?

**Comment by e96wic** [comment]
@katbyte would you review again please?

File: azurerm/resource_arm_monitor_metric_alert.go [hunk 2]
```diff
@@ -37,16 +37,25 @@ func resourceArmMonitorMetricAlert() *schema.Resource {
 				ValidateFunc: validate.NoEmptyStrings,
 			},
 
+			"target_resource_type": {
```
Comment (katbyte) [inline azurerm/resource_arm_monitor_metric_alert.go]: This looks unrelated to removing the max items property? PR title might warrant updating?

File: azurerm/resource_arm_monitor_metric_alert.go (hunk 1 above)
Comment (katbyte) [reply azurerm/resource_arm_monitor_metric_alert.go]: This is a breaking change? And I believe you can do this via CustomizeDiff but i'm not sure how exactly to do it.

File: azurerm/resource_arm_monitor_metric_alert.go [hunk 3]
```diff
@@ -395,11 +431,28 @@ func flattenMonitorMetricAlertCriteria(input insights.BasicMetricAlertCriteria)
...
+
+		for _, val := range *criteria.AllOf {
+			metricCriteria, ok := val.AsMetricCriteria()
+			if !ok {
+				continue
+			}
+			metricData = append(metricData, *metricCriteria)
+		}
+	} else {
+		metricData = *criteria.AllOf
```
Comment (katbyte) [inline azurerm/resource_arm_monitor_metric_alert.go]: I think you'll need the nil checkes here?

File: website/docs/r/monitor_metric_alert.html.markdown [hunk 4]
```diff
@@ -70,6 +70,8 @@ The following arguments are supported:
 * `name` - (Required) The name of the Metric Alert. Changing this forces a new resource to be created.
 * `resource_group_name` - (Required) The name of the resource group in which to create the Metric Alert instance.
 * `scopes` - (Required) The resource ID at which the metric criteria should be applied.
```
Comment (katbyte) [inline website/docs/r/monitor_metric_alert.html.markdown]: Could we update this to reflect multiple can now be used?

**Comment by tombuildsstuff** [comment]
hey @e96wic 

Whilst I'd like to thank you for this contribution - since there's some outstanding comments and we've not heard back from you here, rather than leaving this PR open indefinitely I'm going to close this PR for the moment. That said if you (or someone else) is able to finish this off then we're more than happy to take another look and get this merged :)

Thanks!
